SYCL/ESIMD/api/functional/ctors/common.hpp

```diff
@@ -14,20 +14,44 @@
 #pragma once
 
 #include "../common.hpp"
+#include <string>
 
 namespace esimd_test::api::functional::ctors {
 
-template <typename DataT, int NumElems, typename ContextT>
+#ifdef ESIMD_TESTS_DISABLE_DEPRECATED_TEST_DESCRIPTION_FOR_LOGS
+
+template <int NumElems, typename ContextT>
 class TestDescription : public ITestDescription {
+public:
+  TestDescription(const std::string &data_type) : m_data_type(data_type) {}
+
+  std::string to_string() const override {
+    std::string test_description("simd<");
+
+    test_description += m_data_type + ", " + std::to_string(NumElems) + ">";
+    test_description += ", with context: " + ContextT::get_description();
+
+    return test_description;
+  }
+
+private:
+  const std::string m_data_type;
+};
+
+#else
+
+// Deprecated, use TestDescription<NumElems, ContextT> for new tests instead
+//
+// TODO: Remove deprecated TestDescription from all tests
+template <typename DataT, int NumElems, typename ContextT>
+class [[deprecated]] TestDescription : public ITestDescription {
```

Comment:
@vasilytric This attribute would trigger a compiler warning for every template instantiation by-design. Let me know if that's not needed from your point of view.

Reply:
I think that this attribute will be useful